app/imageUtils.js

```diff
@@ -0,0 +1,208 @@
+        const allImagesSaved = [...webpImages, ...jpegImages].every(result => !!result)
+        return allImagesSaved
+    } catch(err) {
+        logger.info(`Save Image: ${err}`)
+    }
+}
+
+async function createImages(image, imageName, imageWidths, folder, format) {
+    const responses = []
+    for(let width of imageWidths) {
+        const dirPath = getImageDirectory(path.join(folder, width))
+        const hasPermissions = await hasIOPermissions(dirPath)
+        if(!hasPermissions) throw new Error(`Error: Invalid Permissions at ${dirPath}`)
+        responses.push(await __saveImageToFile(image, `${imageName}.${format}`, dirPath, parseInt(width), format))
+    }
+    return responses
+}
+async function __saveImageToFile(image, imageName, dirPath, width, format) {
+    const filePath = path.join(dirPath, imageName)
+    if(format === WEBP) return await sharp(image.data).resize({width}).webp(WEBP_OPTIONS).toFile(filePath)
+    return await sharp(image.data).resize({width}).jpeg(JPEG_OPTIONS).toFile(filePath)
+}
+async function __getImage(res, imageName, folder, webpFormat, width) {
+    try {
+        if(!imageName || !folder) throw new Error('Error: Invalid parameters: ', imageName, folder)
+        const imageFormat = webpFormat ? WEBP : JPEG
+        const mimeType = webpFormat ? WEBP_MIMETYPE : JPEG_MIMETYPE
+
+        const dirPath = getImageDirectory(path.join(folder, width))
+        const hasPerms = await hasIOPermissions(dirPath)
+        if(!hasPerms) throw new Error(`Invalid R/W Permissions at ${dirPath}`)
+        
+        const filePath = path.join(dirPath, `${imageName}.${imageFormat}`)
+        if(!fs.existsSync(filePath)) throw new Error(`Error: ${filePath} does not exist`)
+        const imageBuffer = await fs.promises.readFile(filePath)
+        res.set('Content-Type', mimeType)
+        return res.send(imageBuffer)
+    } catch(err) {
+        logger.info(`__getImage: ${err}`)
+        return res.sendStatus(400)
+    }
+}
+
+async function getArticleImage(res, imageName, webpFormat, width) {
+    try {
+        if(!imageName) throw new Error('GetArticleImages: Invalid Parameters', imageName)
+        return await __getImage(res, imageName, ARTICLE, webpFormat, width)
+    } catch(err) {
+        logger.info(`GetArticleImage ${err}`)
+        return res.sendStatus(500)
+    }
+}
+
+async function setArticleContentImage(imageData) {
+    try {
+        if(!imageData) throw new Error('setContentImage: Invalid Parameters', imageData)
+        const imageName = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 10).concat(`.${JPEG}`)
+        const hasBeenSaved = await __saveImage(imageData, imageName, ARTICLE, CONTENT_IMAGE_SIZES)
+        if(hasBeenSaved) return imageName
+        throw new Error('SetArticleContentImage: Couldn\'t Save Image')
+    } catch(err) {
+        return logger.info(`SetArticleContentImage: ${err}`)
```

Comment:
won't it display stack trace, logging out the function name might be unecessary

Reply:
Not all the time, it makes debugging significantly easier just to add this to the catch logs